rbd/pool_nautilus.go

```diff
@@ -0,0 +1,74 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus is the first release that includes rbd_pool_metadata_get(),
+// rbd_pool_metadata_set() and rbd_pool_metadata_remove().
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <rados/librados.h>
+// #include <rbd/librbd.h>
+// #include <stdlib.h>
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/retry"
+	"github.com/ceph/go-ceph/rados"
+)
+
+// GetPoolMetadata returns pool metadata associated with the given key.
+//
+// Implements:
+//  int rbd_pool_metadata_get(rados_ioctx_t io_ctx, const char *key, char *value, size_t *val_len);
+func GetPoolMetadata(ioctx *rados.IOContext, key string) (string, error) {
```

Comment:
I just reported #299 and it would apply here as well. You may want to validate the passed arguments. Not needed now, as many other functions are affected with this too.

Reply:
Will send a follow-up in some time as this is already approved by two reviewers.